Develop1

Anna-thorn opened the pull request:
Задание 1: юнит-тесты
Покрыть тестами классы Bun, Burger, Ingredient, IngredientType.
Процент покрытия должен быть не ниже 70%.

Commit:
Задание 1: юнит-тесты

Покрыть тестами классы Bun, Burger, Ingredient, IngredientType.

Процент покрытия должен быть не ниже 70%.

Commit:
add jacoco report

okryukova commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Test // проверяет, что конструктор корректно инициализирует объект и устанавливает все необходимые поля
+    public void testConstructor() {
```
⛔️Нужно исправить.  Тест дублирует предыдущие два. Его можно удалить

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,139 @@
+
+        String receipt = burger.getReceipt();
+
+        assertTrue("В чеке нет названия булочки",
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

Commit:
Исправления:
1. В BunTest удален дублирующий тест - testConstructor
2. В BurgerTest тест testGetReceipt разделен на 4 отдельных теста
3. В IngredientTest
     - удален дублирующий тест testIngredientCreation и testIngredientTypeSauce
     - тест testGetters разделен на 4 отдельных теста
     - добавлены вспомогательные методы для создания соуса и начинки
     - переименован testIngredientTypeFilling в testGetTypeFilling

Commit:
add jacoco report2

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,168 @@
+    public void testReplaceBun() {
+        burger.setBuns(mockBun1);
+        burger.setBuns(mockBun2);
+        assertEquals("После замены булочки название не совпало",
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,168 @@
+                receipt.contains("(==== " + TEST_BUN_NAME_1 + " ====)"));
+    }
+    @Test
+    public void testGetReceiptContainsSauceInfo() {
```
⚠️Можно улучшить.  Для проверки рецепта лучше использовать softAssertions

Commit:
Исправления:
1. В pom.xml добавлена зависимость AssertJ (для SoftAssertions)
2. В BurgerTest:
   - вернула старый метод testGetReceipt добавив использование softAssertions
   - удалены: testGetReceiptContainsBunName, testGetReceiptContainsSauceInfo, testGetReceiptContainsFillingInfo, testGetReceiptPriceCalculation
   - в testSetBuns, testReplaceBun, testAddIngredient, testRemoveIngredient, testMoveIngredient использован softAssertions

Commit:
add jacoco report3
